datafusion/expr-common/src/interval_arithmetic.rs

```diff
@@ -286,6 +286,11 @@ impl Interval {
         }
     }
 
+    /// Create a new `Interval` with the same lower and upper bounds.
+    pub fn try_singleton(value: ScalarValue) -> Self {
```

Comment:
I think a better name for this function would be just "singleton". Functions named `try_*` imply they are fallible and generally return a `Result` type.

Reply:
Another idiomatic possibility would be to implement `From<ScalarValue>` for `Interval`.

Reply:
> Another idiomatic possibility would be to implement From<ScalarValue> for Interval.
Sounds good!